From 64d682a646d111a0250bbe3bff77ef0cead91403 Mon Sep 17 00:00:00 2001
From: Hannah Kiekens <hannahkiekens@gmail.com>
Date: Tue, 28 Feb 2023 21:04:37 +0000
Subject: [PATCH] Enable GLSL for Mali (Lima) / PinePhone devices

Commit 88cf8355 changed the behaviour of Mali (Lima) / PinePhone devices by disabling GLSL
88cf8355 got backported in 5.27.1 and broke PinePhone devices (White rectangle on topright quarter of a black screen)

This patch restores the behaviour of 5.27.0


(cherry picked from commit 0543949df709c69c99c46c9abad483250b01c288)
---
 .../libkwineffects/data/glplatform/lima-mali400-desktop-3.0    | 2 +-
 src/libkwineffects/kwinglplatform.cpp                          | 3 +--
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/autotests/libkwineffects/data/glplatform/lima-mali400-desktop-3.0 b/autotests/libkwineffects/data/glplatform/lima-mali400-desktop-3.0
index 8e32a85af81..cf83810fd8b 100644
--- a/autotests/libkwineffects/data/glplatform/lima-mali400-desktop-3.0
+++ b/autotests/libkwineffects/data/glplatform/lima-mali400-desktop-3.0
@@ -6,7 +6,7 @@ ShadingLanguageVersion=1.30
 
 [Settings]
 LooseBinding=true
-GLSL=false
+GLSL=true
 TextureNPOT=true
 Mesa=true
 Lima=true
diff --git a/src/libkwineffects/kwinglplatform.cpp b/src/libkwineffects/kwinglplatform.cpp
index 82554ca755b..aa0f5b105a5 100644
--- a/src/libkwineffects/kwinglplatform.cpp
+++ b/src/libkwineffects/kwinglplatform.cpp
@@ -1138,8 +1138,7 @@ void GLPlatform::detect(OpenGLPlatformInterface platformInterface)
 
     if (isLima()) {
         m_recommendedCompositor = OpenGLCompositing;
-        // GLSL works but causes dramatic FPS drop on this GPU
-        m_supportsGLSL = false;
+        m_supportsGLSL = true;
     }
 
     if (isVideoCore4()) {
-- 
GitLab

